rules-java-ee/addon/src/main/java/org/jboss/windup/rules/apps/javaee/rules/DiscoverJsfSourceFileRuleProvider.java

```diff
@@ -0,0 +1,44 @@
+package org.jboss.windup.rules.apps.javaee.rules;
+
+import org.jboss.windup.config.AbstractRuleProvider;
+import org.jboss.windup.config.GraphRewrite;
+import org.jboss.windup.config.loader.RuleLoaderContext;
+import org.jboss.windup.config.metadata.RuleMetadata;
+import org.jboss.windup.config.operation.iteration.AbstractIterationOperation;
+import org.jboss.windup.config.phase.InitialAnalysisPhase;
+import org.jboss.windup.graph.model.FileLocationModel;
+import org.jboss.windup.graph.service.GraphService;
+import org.jboss.windup.rules.apps.javaee.model.JsfSourceFileModel;
+import org.jboss.windup.rules.files.condition.FileContent;
+import org.ocpsoft.rewrite.config.Configuration;
+import org.ocpsoft.rewrite.config.ConfigurationBuilder;
+import org.ocpsoft.rewrite.context.EvaluationContext;
+
+/**
+ * Discovers JSF files in the input application.
+ *
+ * @author <a href="mailto:jesse.sightler@gmail.com">Jesse Sightler</a>
+ */
+@RuleMetadata(phase = InitialAnalysisPhase.class)
+public class DiscoverJsfSourceFileRuleProvider extends AbstractRuleProvider
+{
+
+    @Override
+    public Configuration getConfiguration(RuleLoaderContext context)
+    {
+        return ConfigurationBuilder.begin()
+                .addRule()
+                .when(FileContent.matches("{taglib1}").inFileNamed("{*}.{extension}"))
+                .perform(new AbstractIterationOperation<FileLocationModel>() {
+                    @Override
+                    public void perform(GraphRewrite event, EvaluationContext context, FileLocationModel payload) {
+                        GraphService.addTypeToModel(event.getGraphContext(), payload.getFile(), JsfSourceFileModel.class);
+                    }
+                })
+                .where("taglib1")
+                    .matches("(java\\.sun\\.com/jsf/core)|(java\\.sun\\.com/jsf/html)|(http://java\\.sun\\.com/jsf/facelets)")
```

Comment:
I know we targets the old technology, but if there is new app analyzed, Btw there are in JSF 2.2 new namespaces like:

- http://java.sun.com/jsf/core-> http://xmlns.jcp.org/jsf/core
- http://java.sun.com/jsf/html -> http://xmlns.jcp.org/jsf/html
- http://java.sun.com/jsf/facelets -> http://xmlns.jcp.org/jsf/facelets
- http://java.sun.com/jsf/composite -> http://xmlns.jcp.org/jsf/composite
- http://java.sun.com/jsp/jstl/core -> http://xmlns.jcp.org/jsp/jstl/core
- http://java.sun.com/jsp/jstl/functions -> http://xmlns.jcp.org/jsp/jstl/functions

Reply:
Good point, I had overlooked these. I will add it tomorrow.

Reply:
I have added the jsf ones, but not the jstl ones. jstl can be used even in non-jsf pages, right?

Reply:
hmm, I overlooked the comment while merging, but JSTL taglibs are for JSPs.

Reply:
I am a little bit confused what we will try to do with JSP as we have `JspSourceFileModel` for them, but you can have JSF code in `*.jsp` file.